lib/postgrex/replication.ex

```diff
@@ -691,13 +705,20 @@ defmodule Postgrex.Replication do
 
   @doc false
   def handle_call({:start_replication, opts}, _from, %{repl_opts: nil, copy_opts: nil} = s) do
+    temporary = Keyword.fetch!(opts, :temporary)
     statement = command(:start_replication, opts)
 
-    with {:ok, protocol} <- Protocol.handle_replication(statement, s.protocol),
+    with :ok <- validate_reconnect_options(s.auto_reconnect, temporary, opts),
+         {:ok, protocol} <- Protocol.handle_replication(statement, s.protocol),
          {:ok, protocol} <- Protocol.checkin(protocol) do
       s = %{s | protocol: protocol, repl_opts: opts}
       {:reply, :ok, s}
     else
+      :error ->
+        # If auto_reconnect is enabled and the temporary option is given,
+        # raise an error if any of the keys in temporary are missing.
+        raise ArgumentError, "expected :plugin and :snapshot to be given in the :temporary option"
```

Comment:
We should validate this in the client, better than crashing in the process later on?

Reply:
Oh, ignore me. We only check this here because we only try to create this slot here if it is temporary. Hrm...

I am thinking that maybe we should change the `start_replication` API slightly. What if instead of `:temporary`, we call it `create_temporary_slot: [...]`, and then it will be applied both on `start_replication` and on auto-reconnect, making sure both the initial behaviour and the reconnect behaviour are the same. The idea is that, if the slot is temporary, it is always automatically started by `start_replication` for you. You don't have to do it yourself. WDYT?

Sorry about the back and forth, but it is a bit tricky to figure out all of the API details.

Reply:
> I am thinking that maybe we should change the start_replication API slightly. What if instead of :temporary, we call it create_temporary_slot: [...], and then it will be applied both on start_replication and on auto-reconnect, making sure both the initial behaviour and the reconnect behaviour are the same. The idea is that, if the slot is temporary, it is always automatically started by start_replication for you. You don't have to do it yourself. WDYT?

I like it. So the idea would be, for both `start_replication` and `copy table`, that you either provide your own permanent slot or provide the options needed to create/recreate a temporary slot. There would be a few other changes to consider in this case:

1. Might want to remove the `temporary` option from `create_slot` because any temporary slot created won't work as expected during auto-reconnect when passed into these commands. It could be considered (and possibly renamed to) `create_permanent_slot`?  If this change isn't made then I think it gets a bit ugly when a user provides a temporary slot. Either:
    - The commands crash upon reconnect
    - We need to make another option for the user to provide the reconnect options
    - We have to get the reconnect options from the `slot_state` query, which wouldn't have the `snapshot` option.

2. `start_replication` and `copy_table` must have 2 options: `slot_name` and `create_temp_slot` and one of them is required but not both. If both are provided then raise?

>Sorry about the back and forth, but it is a bit tricky to figure out all of the API details.

No worries, I agree lol. I second guessed many things while I was writing these. I'm happy to try to make the API better.